napari/utils/transforms/transforms.py

```diff
@@ -148,12 +160,31 @@ def __getitem__(self, key: slice) -> 'TransformChain[_T]':
         ...
 
     def __getitem__(self, value):
-        return super().__getitem__(value)
+        if f"getitem_{value}" not in self._cache_dict:
+            self._cache_dict[f"getitem_{value}"] = super().__getitem__(value)
+        return self._cache_dict[f"getitem_{value}"]
+
+    def __setitem__(self, key, value):
+        super().__setitem__(key, value)
```

Comment:
I should probably know this, but will this call to `super().__setitem__` end up calling `self.__delitem__`?

If not, I think we need to explicitly disconnect the element that's being replaced before calling `super().__setitem__` from `self._clean_cache`.

Reply:
added